Assets/Mirror/Runtime/NetworkReader.cs

```diff
@@ -186,16 +186,14 @@ public ArraySegment<byte> ReadBytesSegment(int count)
         // null support, see NetworkWriter
         public byte[] ReadBytesAndSize()
         {
-            if (ReadBoolean())
-            {
-                return ReadBytes(checked((int)ReadPackedUInt32()));
-            }
-            return null;
+            uint count = ReadPackedUInt32();
```

Comment:
```suggestion
            // count = 0 means the array was null
            // otherwise count -1 is the length of the array 
            uint count = ReadPackedUInt32();
```